packages/squiggle-lang/src/rescript/Distributions/SampleSetDist/KdeLibrary.js

```diff
@@ -1,33 +1,75 @@
-const pdfast = require("pdfast");
 const filter = require("lodash/filter");
 const isFinite = require("lodash/isFinite");
 
-const samplesToContinuousPdf = (
-  samples,
-  size,
-  width,
-  min = false,
-  max = false
-) => {
-  let _samples = filter(samples, isFinite);
-  if (isFinite(min)) {
-    _samples = filter(_samples, (r) => r > min);
-  }
-  if (isFinite(max)) {
-    _samples = filter(_samples, (r) => r < max);
-  }
-
-  // The pdf that's created from this function is not a pdf but a pmf. y values
-  // being probability mass and not density.
-  // This is awkward, because our code assumes later that y is a density
-  let pdf = pdfast.create(_samples, { size, width });
-
-  // To convert this to a density, we need to find the step size. This is kept
-  // constant for all y values
-  let stepSize = pdf[1].x - pdf[0].x;
-
-  // We then adjust the y values to density
-  return { xs: pdf.map((r) => r.x), ys: pdf.map((r) => r.y / stepSize) };
+// Convert samples to x-y pairs for a PDF
+// Uses kernel density estimation (KDE) with a triangular kernel
+// samples: Must be sorted!
+// size:    Number of points in output
+// width:   Width of the kernel, as a whole number of x steps
+const samplesToContinuousPdf = (samples, size, width) => {
+  samples = filter(samples, isFinite); // Not sure if this is needed?
+  const len = samples.length;
+  if (len === 0) return { xs: [], ys: [] };
+
+  // To give room for all the kernels, we'll have width steps
+  // on either side as padding.
+  // Subtract from the size - 1 total intervals.
+  const stepsInside = size - 1 - width * 2;
+  if (stepsInside < 1)
+    throw Error("Number of steps isn't enough to contain all values");
+
+  // Sample min and range, and the size of an x interval dx
+  const smin = samples[0];
+  const srange = samples[len - 1] - smin;
+  const dx = srange / stepsInside;
+
+  // Output min and range
+  // Total number of intervals is:
+  // range / dx === (srange + 2 * margin) / dx
+  //            === stepsInside + 2 * width
+  //            === size - 1
+  const margin = width * dx;
+  const min = smin - margin;
+  const range = srange + 2 * margin;
+
+  // Should catch this case and treat as discrete earlier?
+  if (range === 0) return [{ xs: [min], ys: [1] }];
```

Comment:
Could be good to return something like a NotSupportedError.

Reply:
I've moved the check into the caller. If all the "continuous" samples are equal to each other, then it may as well be treated as a discrete cluster even though the number was less than `minDiscreteWeight`.